staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go

```diff
@@ -292,21 +261,21 @@ func (s *APIAggregator) AddAPIService(apiService *apiregistration.APIService) {
 		serviceResolver: s.serviceResolver,
 	}
 	proxyHandler.updateAPIService(apiService)
+	if err := s.openAPIAggregator.loadApiServiceSpec(proxyHandler, apiService); err != nil {
+		return err
```

Comment:
Unless the call succeed, neither the handler nor its state would change (at least that was my intention). Returning error here will force the controller to retry later.